### ISSUE TITLE
Mapstruct doesn't use custom enum mapper

### ISSUE THREAD

**Issue by nettadhp** [mapstruct/mapstruct#3038]
### Expected behavior

```java
public interface CarMapper {

  CarDto carToCarDto(Car car);

  CarDtoEnum from(CarEnum carEnum);
}
```
should generate something like:

```java
public class CarMapperImpl implements CarMapper {
  @Override
  public CarDto carToCarDto(Car car) {
     ...
      carDto.setCarDtoEnum(from(car.getCarEnum()));
   }
}
```

### Actual behavior

```java
public class CarMapperImpl implements CarMapper {
  @Override
  public CarDto carToCarDto(Car car) {
     ...
      carDto.setCarDtoEnum(Enum.valueOf(CarDtoEnum.class, car.getCarEnum()));
   }
}
```

### Steps to reproduce the problem

see above description.

### MapStruct Version

1.5.2.Final

**Comment by nettadhp** [comment]
Car was a contrived example based off of my code which is quite complicated. I double checked the class and the get returns the Enum not a String.
btw. Enum.valueOf returns the enum.

**Comment by filiphr** [comment]
>btw. Enum.valueOf returns the enum.

Yes `Enum#valueOf` does return the enum. However, it is used to map from `String` into the `CarDtoEnum`. If `Car#getCarEnum` returns `CarEnum` then `Enum.valueOf(CarDtoEnum.class, car.getCarEnum())` will not compile. However, it seems that for you this compiles correctly.

Can you please double check your getters

**Comment by nettadhp** [comment]
I'm going to try to recreate the issue with code that I can post here.

**Comment by filiphr** [comment]
That would be the best @nettadhp. Thanks a lot
